File: src/com/droidwave/stress/Deck.java
package com.droidwave.stress;

import java.util.Collections;
import java.util.List;

public class Deck {

	private List<Integer> cards;

	public Deck(int suitSize, int suitCount) {
		for (int i = 0; i < suitCount; i++) {
			for (int j = 0; j < suitSize; j++) {
				cards.add(j + 1);
			}
		}
		Collections.shuffle(cards);
	}

	public int draw() {
		int cardIndex = (int) Math.random() * cards.size();
		return cards.remove(cardIndex);
	}

	public int size() {
		return cards.size();
	}

	public void add(List<Integer> newDeck) {
		cards.addAll(newDeck);
		Collections.shuffle(cards);
	}

	public void add(int card) {
		cards.add(card);
		return;
	}

}

File: app/src/main/java/com/droidwave/stress/Deck.kt
package com.droidwave.stress

import java.util.ArrayList
import java.util.Random

class Deck {
    private val cards: MutableList<Int> = ArrayList()

    init {
        for (i in 0 until DECK_SUITE_COUNT) {
            for (j in 0 until DECK_SIZE) {
                cards.add(j + 1)
            }
        }
        cards.shuffle()
    }

    internal fun draw(): Int {
        if (cards.isEmpty()) {
            return 0
        }
        val cardIndex = Random().nextInt(cards.size)
        return cards.removeAt(cardIndex)
    }

    internal fun size() = cards.size

    internal fun add(card: Int) {
        cards.add(card)
    }

    companion object {
        val DECK_SUITE_COUNT = 4
        val DECK_SIZE = 13
        val CARDS_IN_DECK = DECK_SUITE_COUNT * DECK_SIZE
    }
}
